src/Service/MailTemplateHelper.php

```diff
@@ -322,17 +325,39 @@ private function getValues(?object $entity, TemplateProcessor $templateProcessor
                 }, $placeHolders)
             );
         } else {
-            // Convert entity to array.
+            $case = $this->getCase($entity);
 
+            // Convert entity to array.
             $data = json_decode($this->serializer->serialize($entity, 'json', ['groups' => ['mail_template']]), true);
 
             // Make hearing, agenda and case data easily available.
             if ($entity instanceof HearingPost) {
                 $data += json_decode($this->serializer->serialize($entity->getHearing(), 'json', ['groups' => ['mail_template']]), true);
+
+                $relation = match (true) {
+                    $entity instanceof HearingPostRequest => $this->relationRepository->findOneBy(['case' => $case->getId()->toBinary(), 'party' => $entity->getRecipient()->getId()->toBinary()]),
+                    $entity instanceof HearingPostResponse => $this->relationRepository->findOneBy(['case' => $case->getId()->toBinary(), 'party' => $entity->getSender()->getId()->toBinary()]),
+                    default => throw new MailTemplateException(sprintf('Invalid entity type %s, expected instance of %s or %s', get_class($entity), HearingPostRequest::class, HearingPostResponse::class)),
+                };
+
+                if (!$relation) {
+                    throw new MailTemplateException('Could not find relation between party and case.');
+                }
+
+                $relationData = json_decode($this->serializer->serialize($relation, 'json', ['groups' => ['mail_template']]), true);
+
+                $data += ['relation' => $relationData];
+
+                switch (get_class($entity)) {
+                    case HearingPostRequest::class:
+                        $data['recipient'] += $relationData;
+                        break;
+                    case HearingPostResponse::class:
+                        $data['sender'] += $relationData;
+                }
```

Comment:
Add a comment on why we're adding relation data in more than one place.

Consider computing a key to add `$relationData` to, e.g.

```php
$key = $entity instanceof HearingPostRequest ? 'recipient' : 'sender';
$data[$key] += $relationData;
```

Reply:
Good idea, fixed.